Ensure we don't remove endpoints that are in use. (#94)
18F/identity-handbook: _articles/annual-certificate-rotation.md

    - Ensure that the new certificate expires on April 1st the next year.
        - `openssl x509 -in saml20xx.crt -text -noout`
2. Update reference to the new 20XX cert in secrets saml_endpoint_configs:
    - You will need to append the new certificate to saml_endpoint_configs in the secrets file in `dev`, `int`, `staging` and `prod` environments.
        - Do not update the configs until you have confirmed that the certs have been pushed to instances in that env (see above).
    - Get access to the new certificate's passphrase from the engineer who generated the new certs (e.g., via a shared Google doc)
    - Edit the env configs via `bin/app-s3-secret --env <ENV> --app idp --edit` in the `identity-devops` repo.
    - So this:
	 ```yaml
	 saml_endpoint_configs: '[{"suffix":"2021","secret_key_passphrase":"XXXXXXXXXXXX"}]'
	 ```
    - Becomes:
	 ```yaml
	 saml_endpoint_configs: '[{"suffix":"2021","secret_key_passphrase":"XXXXXXXXXXXX"},{"suffix":"2022","secret_key_passphrase":"XXXXXXXXXXXX"}]'
	 ```
    - Recycle the env to make new configs take effect.
3. Confirm the new endpoint is live by going to `/api/saml/metadata20XX` in that environment's idp.

> 3. New SAML 20XX has been tested using a new test app on the Dashboard

Reviewer: *Nitpick:* I think Markdown can handle it but this should be `4` and the rest of the numbers updated.

Technology Transformation Services
U.S. General Services Administration
[partners@login.gov](mailto:partners@login.gov)
</blockquote>

> ## Steps to remove and old SAML certificate:

Reviewer 2: ```suggestion
## Steps to remove an old SAML certificate:
```

> 1. Run a CloudWatch query to ensure that partners are no longer the endpoint for the year being retired, e.g.,

Reviewer 2: ```suggestion
1. Run a CloudWatch query to ensure that partners are no longer using the endpoint for the year being retired:
```

    ```
    fields
      substr(replace(path, "/api/saml/auth", ""), 0, 4) AS year,
      @timestamp, @message
    | sort @timestamp desc
    | filter path like "/api/saml/auth"

>     | stats count(*) by year

Reviewer 3: let's comment the file we read from. This is `production.log` right?

Also I was able to get a version of the query that included HTTP referer, let's add this too?

```
# file prod_/var/log/nginx/access.log
fields @timestamp, @message
| parse @message 'http_referer="*"' as http_referer
| parse @message 'uri_path="*"' as path
| sort @timestamp desc
| filter path like "/api/saml/auth"
| filter http_referer not like "secure.login.gov"
| limit 20
| display
  replace(path, "/api/saml/auth", "") AS year,
  http_referer AS http_referer,
  @timestamp
| stats count(*) by year, http_referer
```

Author: Added in f84969e

    ```

> 2. If the endpoint is no longer in use, remove reference to the old 20XX cert in secrets saml_endpoint_configs:

Reviewer 2: ```suggestion
2. If the endpoint is no longer in use, remove references to the old 20XX cert in secrets saml_endpoint_configs:
```
